packages/wallet/backend/tests/grant/controller.test.ts

```diff
@@ -0,0 +1,184 @@
+import { Container } from '@/shared/container'
+import { Bindings } from '@/app'
+import { createApp, TestApp } from '@/tests/app'
+import { Knex } from 'knex'
+import { AuthService } from '@/auth/service'
+import {
+  createRequest,
+  createResponse,
+  MockRequest,
+  MockResponse
+} from 'node-mocks-http'
+import { Request, Response } from 'express'
+import { mockLogInRequest } from '@/tests/mocks'
+import { GrantController } from '@/grant/controller'
+import { applyMiddleware } from '@/tests/utils'
+import { withSession } from '@/middleware/withSession'
+import { User } from '@/user/model'
+import { faker } from '@faker-js/faker'
+import { createContainer } from '@/createContainer'
+import { env } from '@/config/env'
+import { createUser } from '@/tests/helpers'
+import { truncateTables } from '@/tests/tables'
+import { errorHandler } from '@/middleware/errorHandler'
+
+describe('Grant Controller', () => {
+  let bindings: Container<Bindings>
+  let appContainer: TestApp
+  let knex: Knex
+  let authService: AuthService
+  let grantController: GrantController
+  let req: MockRequest<Request>
+  let res: MockResponse<Response>
+
+  const next = jest.fn()
+  const args = mockLogInRequest().body
+
+  const createReqRes = async () => {
+    res = createResponse()
+    req = createRequest()
+
+    await applyMiddleware(withSession, req, res)
+
+    const { user, session } = await authService.authorize(args)
+    req.session.id = session.id
+    req.session.user = {
+      id: user.id,
+      email: user.email,
+      needsWallet: !user.rapydWalletId,
+      needsIDProof: !user.kycId
+    }
+    await User.query().patchAndFetchById(user.id, { rapydWalletId: 'mocked' })
+  }
+
+  const createMockGrantControllerDeps = (isFailure?: boolean) => {
+    const grantControllerDepsMocked = {
+      rafikiAuthService: {
+        listGrants: jest.fn().mockReturnValue(['grant1']),
```

Comment:
I think it would be better mock this as an array of grants instead of just an array of strings.